core/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -543,6 +601,14 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeOptionalString(pipeline);
         out.writeBoolean(isRetry);
         out.writeLong(autoGeneratedTimestamp);
+        if (out.getVersion().after(Version.V_5_3_0_UNRELEASED)) { // TODO update to onOrAfter after backporting
+            if (contentType != null) {
```

Comment:
that seems wrong. we always make the assumption that serialization relies on the request being valid according to validate I think. Let's fix as a followup though. Open an issue? (If you agree that this should be fixed of course)